api/src/main/java/org/apache/iceberg/FileFormat.java

```diff
@@ -25,23 +25,29 @@
  * Enum of supported file formats.
  */
 public enum FileFormat {
-  ORC("orc", true),
-  PARQUET("parquet", true),
-  AVRO("avro", true),
-  METADATA("metadata.json", false);
+  ORC("orc", true, true),
+  PARQUET("parquet", true, true),
+  AVRO("avro", true, false),
+  METADATA("metadata.json", false, false);
 
   private final String ext;
   private final boolean splittable;
+  private final boolean offsets;
 
-  FileFormat(String ext, boolean splittable) {
+  FileFormat(String ext, boolean splittable, boolean offsets) {
     this.ext = "." + ext;
     this.splittable = splittable;
+    this.offsets = offsets;
   }
 
   public boolean isSplittable() {
     return splittable;
   }
 
+  public boolean hasOffsets() {
+    return offsets;
```

Comment:
Don't we want to treat every file the same? If it has split offsets, we use them and if not we try to split at the target size? That is dependent on the file metadata and not the format.

Reply:
This seemed much more confusing to me because of the split behavior on files without split offset information. I'd rather it's just explicit, we have files we know how to split but can only be split on offsets, and we have files that we know how to split can be split arbitrarily. I was trying to figure out a another term for this, basically I want to differentiate between formats like ORC , Parquet and Avro

Reply:
To be clear, this is not treating every file the same but saying that this file has the potential of having offsets as opposed to being arbitrarily splittable.